javascript/javascript2/week3/index.js

```diff
@@ -0,0 +1,213 @@
+// 3. Create a button in html. When clicking this button,
+// use the function you created in the previous task to log out the text:
+// Called after 5 seconds 5 seconds after the button is clicked.
+
+const delayButton = document.getElementById("delay-button");
+const locationButton = document.getElementById("location-button");
+
+delayButton.addEventListener("click", () => {
+  delayedLogout("Called after 5 seconds", delayTime);
+});
+
+/////////////////////////////////////////////////////////////////////////
+// 4. Create two functions and assign them to two different variables.
+// One function logs out Earth, the other function logs out Saturn.
+// Now create a new third function that has one parameter: planetLogFunction.
+//  The only thing the third function should do is call the provided parameter function.
+// Try call the third function once with the Earth function and once with the Saturn function.
+
+let earthLogger = () => {
+  console.log("earth");
+};
+
+let saturnLogger = () => {
+  console.log("saturn");
+};
+
+thirdFunction = (planetLogFunction) => {
+  planetLogFunction();
+};
+
+thirdFunction(earthLogger);
+thirdFunction(saturnLogger);
+
+/////////////////////////////////////////////////////////////////////////
+// 5. Create a button with the text called "Log location".
+// When this button is clicked the location latitude, longitude of the user
+// should be logged out.
+
+let currentLatitude;
+let currentLongtitude;
+let accuracy;
+
+locationButton.addEventListener("click", () => {
+  getLocation();
+});
+
+function getLocation() {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(createCoordinateList);
+  } else {
+    console.log("Geolocation is not supported by this browser.");
+  }
+}
+
+function createCoordinateList(currentPosition) {
+  currentLatitude = currentPosition.coords.latitude;
+  const coordinateList = document.getElementById("coordinate-list");
+  const coordinateListItemLatitude = document.createElement("LI");
+  coordinateListItemLatitude.innerHTML = `Your latitude : ${currentLatitude}`;
+  coordinateList.appendChild(coordinateListItemLatitude);
+
+  currentLongtitude = currentPosition.coords.longitude;
+  const coordinateListItemLongtitude = document.createElement("LI");
+  coordinateListItemLongtitude.innerHTML = `Your longtitude : ${currentLongtitude}`;
+  coordinateList.appendChild(coordinateListItemLongtitude);
+
+  accuracy = currentPosition.coords.accuracy;
+  const coordinateListItemAccuracy = document.createElement("LI");
+  coordinateListItemAccuracy.innerHTML = `The accuracy : ${accuracy}`;
+  coordinateList.appendChild(coordinateListItemAccuracy);
+
+  showGoogleMaps(currentLatitude, currentLongtitude);
+}
+
+/////////////////////////////////////////////////////////////////////////
+// 6. Optional Now show that location on a map using fx the Google maps api
+
+function showGoogleMaps(latitude, longitude) {
```

Comment:
this part does not seem to work. 

I tried this solution  https://developers.google.com/maps/documentation/javascript/adding-a-google-map#maps_add_map-javascript

Reply:
https://maps.googleapis.com/maps/api/js/AuthenticationService.Authenticate?1shttp://127.0.0.1:5500/index.html&4sKEYHASBEENREMOVEDBYME&callback=_xdc_._6ntu5o&key=KEYHASBEENREMOVEDBYME&token=104540

==>> The reason why it doesn't want to work cause I used my own google API key when I was writing the code.
For obvious reasons I have not published my API key here on Github.  ==> KEYHASBEENREMOVEDBYME